### PR TITLE
Use RawGit to fetch list generators stored on github

### PR DESCRIPTION
**Pull request by zckrs**
Related to https://github.com/yeoman/yeoman-generator-list/pull/52

### CODE REVIEW

File: app/assets/js/generator.js [hunk 1]
```diff
@@ -19,36 +19,52 @@
...
-        });
-
-        if (list.listContainer) {
-          list.on('updated', function () {
-            $('.table thead').toggle(list.matchingItems.length !== 0);
-            $('#search-notfound').toggle(list.matchingItems.length === 0);
+    var lastSha = win.localStorage.getItem('sha');
+    var updatedAt = win.localStorage.getItem('updatedAt');
+
+    if (lastSha && updatedAt && Date.now() - updatedAt < 1800000) {
```
Comment (SBoudrias) [inline app/assets/js/generator.js]: let's give a name to that magic number.

File: app/assets/js/generator.js [hunk 2]
```diff
@@ -19,36 +19,52 @@
...
-            'downloads',
-            'description'
-          ]
-        });
-
-        if (list.listContainer) {
-          list.on('updated', function () {
-            $('.table thead').toggle(list.matchingItems.length !== 0);
-            $('#search-notfound').toggle(list.matchingItems.length === 0);
+    var lastSha = win.localStorage.getItem('sha');
```
Comment (SBoudrias) [inline app/assets/js/generator.js]: I know localStorage is super cool and all, but IMO that could be done pretty easily with a cookie. You'd have 100% browser support + automatic cache cleaning.

File: app/assets/js/generator.js [hunk 3]
```diff
@@ -19,36 +19,52 @@
...
+            });
+          }
         }
-      }
-      else {
-        pluginsAll.text('There was a problem fetching the generator list. Please try again later');
-      }
-    });
+        else {
+          pluginsAll.text('There was a problem fetching the generator list. Please try again later');
```
Comment (SBoudrias) [inline app/assets/js/generator.js]: How about linking to `npms.io` when we fail to serve the list. At least it'd give a fallback to our users.

**Comment by zckrs** [comment]
The 60/hour rate limit is per user.
>For unauthenticated requests, the rate limit allows you to make up to 60 requests per hour. Unauthenticated requests are associated with your IP address

File: app/assets/js/generator.js (hunk 3 above)
Comment (zckrs) [inline app/assets/js/generator.js]: Done
`      pluginsAll.html('There was a problem fetching the generator list. Please try again later.<br />Or you can use <a href="https://npms.io/search?q=keywords%3Ayeoman-generator">npms.io</a>.');
`

**Comment by zckrs** [comment]
Commits squashed

**Comment by SBoudrias** [comment]
The list is down again, I'll merge this PR now and see how it holds off in production after a few days.

**Comment by SBoudrias** [comment]
Good news is it's working.

**Comment by zckrs** [comment]
🎆🎉🎆

**Pull request merged**